src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -139,6 +140,13 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSp
                 var snapshot = firstSpan.Snapshot;
                 Debug.Assert(snapshot.TextBuffer == _subjectBuffer);
 
+                // The LSP client will handle producing tags when running under the LSP editor.
+                // Our tagger implementation should return nothing to prevent conflicts.
+                if (snapshot.TextBuffer.IsInLspEditorContext())
```

Comment:
You _might_ be able to move this into the tagger provider's create tagger method, similar to what you did for the SuggestedActionsSourceProvider.cs. That'll ensure the tagger isn't even created in the first place which might be a better test that we're not depending on it.